File: SozlukJava/app/src/main/java/com/example/sozlukjava/view/ProfilFragment.java
package com.example.sozlukjava.view;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.sozlukjava.R;
import com.example.sozlukjava.adapter.EntryAdapter;
import com.example.sozlukjava.databinding.FragmentProfilBinding;
import com.example.sozlukjava.util.SessionManager;
import com.example.sozlukjava.viewmodel.ProfilViewModel;

// ProfilFragment.java
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SearchView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;

public class ProfilFragment extends Fragment {
    private FragmentProfilBinding binding;
    private ProfilViewModel viewModel;
    private SessionManager session;
    private EntryAdapter recyclerAdapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentProfilBinding.inflate(inflater, container, false);
        binding.btnCikisYap.setOnClickListener(v ->
                new AlertDialog.Builder(requireContext())
                        .setTitle("Çıkış Yap")
                        .setMessage("Oturumunuzu kapatmak istediğinizden emin misiniz?")
                        .setPositiveButton("Evet", (dialog, which) -> {
                            session.clearSession();
                            NavHostFragment.findNavController(ProfilFragment.this)
                                    .navigate(R.id.action_profilFragment_to_girisFragment);
                        })
                        .setNegativeButton("İptal", null)
                        .show()
        );
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        session = new SessionManager(requireContext());
        viewModel = new ViewModelProvider(this).get(ProfilViewModel.class);

        int userId = session.getUserId();
        String username = session.getUsername();
        binding.tvKullaniciAdi.setText(username != null ? username : "Bilinmeyen Kullanıcı");

        binding.profilRecyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));

        viewModel.loadUserEntries(userId);

        binding.bottomNavBugun.setSelectedItemId(R.id.nav_profil);
        binding.bottomNavBugun.setOnItemSelectedListener(item -> {
            int id = item.getItemId();
            if (id == R.id.nav_gundem) {
                NavHostFragment.findNavController(this).navigate(R.id.anaSayfaFragment);
                return true;
            } else if (id == R.id.nav_bugun) {
                NavHostFragment.findNavController(this).navigate(R.id.bugunFragment);
                return true;
            } else if (id == R.id.nav_profil) {
                return true;
            }
            return false;
        });

        recyclerAdapter = new EntryAdapter(
                new java.util.ArrayList<>(),
                entry -> {
                    ProfilFragmentDirections.ActionProfilFragmentToEntryDetayFragment action =
                            ProfilFragmentDirections.actionProfilFragmentToEntryDetayFragment(entry.getId());
                    NavHostFragment.findNavController(ProfilFragment.this).navigate(action);
                },
                entry -> new AlertDialog.Builder(requireContext())
                        .setTitle("Entry Sil")
                        .setMessage("Bu entry'i silmek istediğinizden emin misiniz?")
                        .setPositiveButton("Evet", (d, w) -> viewModel.deleteEntry(entry.getId(), userId))
                        .setNegativeButton("İptal", null)
                        .show()
        );
        binding.profilRecyclerView.setAdapter(recyclerAdapter);

        viewModel.getEntries().observe(getViewLifecycleOwner(), list -> {
            // basitçe yeni adapter verebilir veya diff util kullanabilirsiniz
            binding.profilRecyclerView.setAdapter(
                    new EntryAdapter(list,
                            entry -> {
                                ProfilFragmentDirections.ActionProfilFragmentToEntryDetayFragment action =
                                        ProfilFragmentDirections.actionProfilFragmentToEntryDetayFragment(entry.getId());
                                NavHostFragment.findNavController(ProfilFragment.this).navigate(action);
                            },
                            entry -> new AlertDialog.Builder(requireContext())
                                    .setTitle("Entry Sil")
                                    .setMessage("Bu entry'i silmek istediğinizden emin misiniz?")
                                    .setPositiveButton("Evet", (d, w) -> viewModel.deleteEntry(entry.getId(), userId))
                                    .setNegativeButton("İptal", null)
                                    .show()
                    )
            );
        });

        viewModel.deleteResult.observe(getViewLifecycleOwner(), it -> {
            if (it != null && it.isSuccess()) {
                Toast.makeText(requireContext(), "Silindi", Toast.LENGTH_SHORT).show();
            } else {
                String msg = (it != null && it.getMessage() != null) ? it.getMessage() : "Silinemedi";
                Toast.makeText(requireContext(), msg, Toast.LENGTH_SHORT).show();
            }
        });

        binding.searchViewProfil.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override public boolean onQueryTextSubmit(String query) {
                viewModel.setSearchQuery(query != null ? query : "");
                return true;
            }
            @Override public boolean onQueryTextChange(String newText) {
                viewModel.setSearchQuery(newText != null ? newText : "");
                return true;
            }
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: SozlukKotlin/app/src/main/java/com/example/sozlukkotlin/view/ProfilFragment.kt
package com.example.sozlukkotlin.view

import android.app.AlertDialog
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.SearchView
import android.widget.Toast
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import com.example.sozlukkotlin.R
import com.example.sozlukkotlin.adapter.EntryAdapter
import com.example.sozlukkotlin.databinding.FragmentProfilBinding
import com.example.sozlukkotlin.model.Entry
import com.example.sozlukkotlin.util.SessionManager
import com.example.sozlukkotlin.viewmodel.ProfilViewModel
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach

class ProfilFragment : Fragment() {
    private var _binding: FragmentProfilBinding? = null
    private val binding get() = _binding!!

    private lateinit var viewModel: ProfilViewModel
    private lateinit var session: SessionManager
    private lateinit var recyclerAdapter: EntryAdapter
    private var entryList: List<Entry> = listOf()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        _binding = FragmentProfilBinding.inflate(inflater, container, false)
        binding.btnCikisYap.setOnClickListener {
            AlertDialog.Builder(requireContext())
                .setTitle("Çıkış Yap")
                .setMessage("Oturumunuzu kapatmak istediğinizden emin misiniz?")
                .setPositiveButton("Evet") { _, _ ->
                    session.clearSession()
                    findNavController().navigate(R.id.action_profilFragment_to_girisFragment)
                }
                .setNegativeButton("İptal", null)
                .show()
        }
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        session = SessionManager(requireContext())
        viewModel = ViewModelProvider(this)[ProfilViewModel::class.java]

        val userId = session.getUserId()
        val username = session.getUsername()

        binding.tvKullaniciAdi.text = username ?: "Bilinmeyen Kullanıcı"

        viewModel.loadUserEntries(userId)
        binding.bottomNavBugun.selectedItemId = R.id.nav_profil
        binding.bottomNavBugun.setOnItemSelectedListener {
            when (it.itemId) {
                R.id.nav_gundem -> { findNavController().navigate(R.id.anaSayfaFragment); true }
                R.id.nav_bugun ->{ findNavController().navigate(R.id.bugunFragment); true }
                R.id.nav_profil -> true

                else -> false
            }
        }

        recyclerAdapter = EntryAdapter(entryList,
            onClick = { entry ->
                val action = ProfilFragmentDirections.actionProfilFragmentToEntryDetayFragment(entry.id)
                findNavController().navigate(action)
            },
            onLongClick = { entry ->
                AlertDialog.Builder(requireContext())
                    .setTitle("Entry Sil")
                    .setMessage("Bu entry'i silmek istediğinizden emin misiniz?")
                    .setPositiveButton("Evet") { _, _ ->
                        viewModel.deleteEntry(entry.id, userId)
                    }
                    .setNegativeButton("İptal", null)
                    .show()
            }
        )
        binding.profilRecyclerView.adapter = recyclerAdapter
        viewModel.deleteResult.observe(viewLifecycleOwner) {
            if (it.success) {
                Toast.makeText(requireContext(), "Silindi", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(requireContext(), it.message ?: "Silinemedi", Toast.LENGTH_SHORT).show()
            }
        }
        binding.searchViewProfil.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String?): Boolean {
                viewModel.setSearchQuery(query.orEmpty())
                return true
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                viewModel.setSearchQuery(newText.orEmpty())
                return true
            }
        })

    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
